cheri_compressed_cap_common.h

```diff
@@ -67,11 +67,20 @@ _CC_STATIC_ASSERT_SAME(_CC_N(FIELD_INTERNAL_EXPONENT_SIZE) + _CC_N(FIELD_EXP_NON
                            _CC_N(FIELD_EXPONENT_LOW_PART_SIZE),
                        _CC_N(FIELD_EBT_SIZE));
 
+// In order to allow vector loads and store from memory we can optionally reverse the first two fields.
 struct _cc_N(cap) {
     /* offset = cursor - base */
+#ifdef _CC_CAP_MEMORY_ORDER_BIG_ENDIAN
```

Comment:
Hm, isn't the reversal a property of both the host and the target? I.e. this really should have been left as a BLA_REVERSE macro, since big-endian hosts would want to reverse for little-endian targets when doing 128-bit loads and not reverse for big-endian targets.

Reply:
I guess that depends whether the QEMU load is a 128-bit integer load or a 2xi64 load?

Reply:
Yeah, hence why it should be given a less specific name so the consumer can choose when it's needed (and is also clearer that it isn't _required_)